File: examples/first_10_subjects.js
#!/usr/bin/env node

var zerorpc = require("zerorpc");
var client = new zerorpc.Client();

// var config = require('../config.json');
// log_in(config.username, config.password);
var API_SERVER_LOC = 'tcp://0.0.0.0:9999';

client.connect(API_SERVER_LOC);

var getSubjects = function(n) {

  client.invoke('first_n_subjects', n, function(error, response, stream) {

    console.log('\nThe first 10 emails in your inbox...\n');

    if (! response instanceof Array) {
        throw new Error('Response should be an array');
    }

    for (var i=0; i < response.length; i++) {
        var subject = response[i][0]; // first item

        console.log('    ' + subject);
    }

    process.exit();
  });
};

getSubjects(10);

File: examples/first_10_subjects.py
#!/usr/bin/env python

import zerorpc

# from inbox.server.config import config, load_config
# load_config()

API_SERVER_LOC = 'tcp://0.0.0.0:9999'
# API_SERVER_LOC = config.get('API_SERVER_LOC', None)

def get_subjects(n):
    api_client = zerorpc.Client(timeout=5)
    api_client.connect(API_SERVER_LOC)

    namespace_id = 2
    subjects = api_client.first_n_subjects(namespace_id, 10)

    print """
The first {0} emails in your inbox...
    """.format(n)

    for s in subjects:
    	print '    ', s[0]

get_subjects(10)
